server/controllers/logstreaming_controller.go

```diff
@@ -114,17 +114,17 @@ func (j *JobsController) GetProjectJobs(w http.ResponseWriter, r *http.Request)
 }
 
 func (j *JobsController) GetProjectJobsWS(w http.ResponseWriter, r *http.Request) {
```

Comment:
Why not just instrument this? like this:
https://github.com/lyft/atlantis/blob/release-v0.17.1-lyft.1/server/events/vcs/instrumented_client.go

I think it's better practice for keeping stats out of core logic.  Sorry If i should have clarified this earlier.  Also tracking latencies as well would be a good standard to keep.

Reply:
Hmm.. the methods here don't return anything. So, I'm not sure how I should use the decorator pattern here to track for failures. Also, when you say tracking latencies, do you mean the `GetProjectJobsWS` method or the `websocket.WriteMessage()` method?

Reply:
Added metric to track web socket write latency.

Reply:
I mean tracking api level metrics and yeah the decorator with interfaces wouldn't work, but you could use function decoration here. Just move all the core logic to another function and when there is an error you can increment the metric once.

Reply:
Done!